### PR TITLE
Modernize ranges::count

### PR DESCRIPTION
**Pull request by miscco**
This applies the new machinery to ranges::count

### CODE REVIEW

File: stl/inc/algorithm [hunk 1]
```diff
@@ -606,26 +606,30 @@ namespace ranges {
...
+        template <input_range _Rng, class _Ty, class _Pj = identity>
+            requires indirect_binary_predicate<ranges::equal_to, projected<iterator_t<_Rng>, _Pj>, const _Ty*>
+        _NODISCARD constexpr range_difference_t<_Rng> operator()(_Rng&& _Range, const _Ty& _Val, _Pj _Proj = {}) const {
+            return _Count_unchecked(_Ubegin(_Range), _Uend(_Range), _Val, _Pass_fn(_Proj));
+        }
+        // clang-format on
+    private:
+        template <class _It, class _Se, class _Ty, class _Pj>
+        _NODISCARD constexpr iter_difference_t<_It> _Count_unchecked(
+            _It _First, const _Se _Last, const _Ty& _Val, _Pj _Proj) const {
```
Comment (CaseyCarter) [inline stl/inc/algorithm:622]: Can we `_STL_INTERNAL_STATIC_ASSERT` the requirements here? (I know it seems silly, but I want to ensure these internal functions are in some way guarded against being used incorrectly under future maintenance. It's an attempt to assuage my guilt over stripping away perfectly good constraints. ;))

File: tests/std/tests/P0896R4_ranges_alg_count/test.cpp [hunk 2]
```diff
@@ -2,52 +2,40 @@
...
-        (void) ranges::count(in, value);
-
-        struct type {
-            bool operator==(type const&) const = default;
-        };
-        using Projection = type (*)(std::iter_common_reference_t<ranges::iterator_t<In>>);
-        (void) ranges::count(in, type{}, Projection{});
+    static constexpr P input[5] = {{0, 99}, {1, 47}, {2, 99}, {3, 47}, {4, 99}};
+
+    template <class Read>
```
Comment (CaseyCarter) [inline tests/std/tests/P0896R4_ranges_alg_count/test.cpp:18]:
```suggestion
    template <ranges::input_range Read>
```

File: tests/std/tests/P0896R4_ranges_alg_count/test.cpp [hunk 3]
```diff
@@ -2,52 +2,40 @@
...
+    static constexpr void call() {
+        using ranges::count;
+        { // Validate iterator + sentinel overload
+            Read wrapped_input{input};
+
+            auto result = count(wrapped_input.begin(), wrapped_input.end(), 47, get_second);
+            STATIC_ASSERT(same_as<decltype(result), iter_difference_t<decltype(wrapped_input.begin())>>);
```
Comment (CaseyCarter) [inline tests/std/tests/P0896R4_ranges_alg_count/test.cpp:25]: Why not use `range_difference_t` here as on 32?

File: stl/inc/algorithm (hunk 1 above)
Comment (miscco) [reply stl/inc/algorithm:622]: Done

File: stl/inc/algorithm [hunk 4]
```diff
@@ -606,26 +606,33 @@ namespace ranges {
...
```
Comment (CaseyCarter) [inline stl/inc/algorithm:622]:
```suggestion
        _NODISCARD static constexpr iter_difference_t<_It> _Count_unchecked(
            _It _First, const _Se _Last, const _Ty& _Val, _Pj _Proj) {
```

**Comment by StephanTLavavej** [comment]
One ranges modernization! Two ranges modernizations! Three ranges modernizations! *Four* ranges modernizations! Ah ah ah ah! :vampire:

**Pull request merged**